### EDA (Exploratory Data Analysis)

In [2]:
import pandas as pd, numpy as np 

In [3]:
pd.set_option('display.max_columns',None)

In [4]:
df = pd.read_csv('cleaned_movie_dataset.csv')

In [5]:
df.head()

,MovieID,Title,Genre,ReleaseYear,ReleaseDate,Country,BudgetUSD,US_BoxOfficeUSD,Global_BoxOfficeUSD,Opening_Day_SalesUSD,One_Week_SalesUSD,IMDbRating,RottenTomatoesScore,NumVotesIMDb,NumVotesRT,Director,LeadActor,ReleaseMonth,ReleaseDay,IsBlockbuster,ProfitUSD,ROI_pct,OpeningWeekRatio,Decade,AverageScore,BudgetTier
0,1,Might Toward Capital,Comedy,2003,2003-09-28,China,6577427.79,6613685.82,15472035.66,1778530.85,3034053.32,6.2,58,7865,10596,Kristina Moore,Brian Mccormick,9,Sunday,False,8894607.87,135.23,0.196,2000s,60.0,High
1,2,He However Experience,Comedy,1988,1988-02-14,Usa,1883810.10,1930949.15,3637731.12,247115.74,831828.84,5.2,44,1708,220,Benjamin Hudson,Ashley Pena,2,Sunday,False,1753921.02,93.10,0.229,1980s,48.0,Mid
2,3,Star Responsibility Politics,Comedy,1971,1971-11-02,Usa,2468079.29,4186694.69,7165111.24,878453.95,2171405.93,5.5,55,4678,7805,Kayla Young,Alexander Haley,11,Tuesday,False,4697031.95,190.31,0.303,1970s,55.0,Mid
3,4,Exactly Live,Comedy,1998,1998-08-06,Usa,1447311.46,2023683.92,4373820.26,570657.72,898886.01,7.3,87,2467,1751,Michael Ross,Patrick Barnett,8,Thursday,False,2926508.80,202.20,0.206,1990s,80.0,Mid
4,5,Focus Improve Especially,Documentary,2021,2021-12-17,India,900915.86,2129629.10,3113017.38,361189.37,861775.91,6.1,67,5555,697,Faith Franklin,Duane Fletcher Dds,12,Friday,False,2212101.52,245.54,0.277,2020s,64.0,Low


In [8]:
# Whcih Genre is most profitable on an average?

genre_avg = df.groupby('Genre',observed = True ).agg(
    avg_profit=('ProfitUSD','mean'),
    avg_ROI = ('ROI_pct','mean'),
    count = ('MovieID','count')    
).round(2).sort_values('avg_ROI',  ascending = False)
print("_____Most Profitable Genre (by Average ROI%)______ ")
print(genre_avg)

_____Most Profitable Genre (by Average ROI%)______ 
              avg_profit  avg_ROI   count
Genre                                    
Horror       17601308.24   178.64  100010
Action       17228120.09   178.40  150131
Comedy       17393408.10   177.93  199832
Sci-Fi       17193285.20   177.72   49802
Thriller     17546817.96   177.54  100071
Drama        17358261.61   177.52  250018
Documentary  17650671.30   177.51   50114
Romance      17438940.46   177.43  100021


**Genre profitability** — Horror leads with the highest ROI% (178.64%), followed by Action (178.40%) and Comedy (177.93%). However, the spread is very narrow, with all genres hovering between 177% and 179% ROI.

In [5]:
# Does budget correlate with box office?
corr_budget_USbo = df['BudgetUSD'].corr(df['US_BoxOfficeUSD'])
corr_budget_Gbo =  df['BudgetUSD'].corr(df['Global_BoxOfficeUSD'])
corr_budget_RT = df['BudgetUSD'].corr(df['RottenTomatoesScore'])
corr_budget_IMDb =  df['BudgetUSD'].corr(df['IMDbRating'])
print("---Budget Correlation---")
print(f"  Budget -> US BO:       {corr_budget_USbo:.4f}")
print(f"  Budget -> Global BO:   {corr_budget_Gbo:.4f}")
print(f"  Budget -> RT Rating:   {corr_budget_RT:.4f}")
print(f"  Budget -> IMDb Rating: {corr_budget_IMDb:.4f}")

---Budget Correlation---
  Budget -> US BO:       0.8827
  Budget -> Global BO:   0.8955
  Budget -> RT Rating:   0.0003
  Budget -> IMDb Rating: 0.0011


**Budget Correlation** — Budget and Global Box Office have very strong correlation of 0.8955 - Higher budget almost means higher box office revenue 

In [8]:
# Which Decade had best avg ROI?

decade_avg = df.groupby('Decade', observed = True).agg(
    avg_roi = ('ROI_pct','mean')
).round(2).sort_values('avg_roi',ascending = False)
print("---Best ROI by Decade---")
print(decade_avg)

---Best ROI by Decade---
        avg_roi
Decade         
1950s    178.47
1960s    178.33
1980s    178.10
2010s    178.05
2000s    177.85
2020s    177.60
1970s    177.52
1990s    177.41


**Best Decade** — The 1950s had the highest Avg ROI (178.47%). But, the difference across all decade is marginal. No single era dramatically outperform another

In [9]:
# Does opening day predict total box office?

corr_opening_USbo = df['Opening_Day_SalesUSD'].corr(df['US_BoxOfficeUSD'])
corr_opening_Gbo = df['Opening_Day_SalesUSD'].corr(df['Global_BoxOfficeUSD'])
print('---Opening day correlation---')
print(f'Opening day -> US BO :     {corr_opening_USbo:.3f}')
print(f'Opening Day -> Global BO : {corr_opening_Gbo:.3f}')

---Opening day correlation---
Opening day -> US BO :     0.955
Opening Day -> Global BO : 0.941


**Opening Day Correlation** — Opening day Sales vs Global Box Office correlation is *0.941* - is the strongest relationship in the entire dataset. A movies's very first day of sales is the best predictor of its total lifetime earning.

In [23]:
# Which country produces high rated movies?

HR_movies = df.groupby('Country' , observed = True).agg(
    avg_imdb = ('IMDbRating','mean'),
    count=('MovieID','count')
).round(2).sort_values('avg_imdb',ascending= False)

print("---IMDb rating by country")
print(HR_movies)

---IMDb rating by country
             avg_imdb   count
Country                      
Japan            6.51   20222
Australia        6.50   30587
Uk               6.50   51277
China            6.50   30952
Usa              6.50  714581
France           6.49   30471
India            6.49   50984
Canada           6.48   40528
Germany          6.48   20211
South Korea      6.48   10186


**Country And Rating** — Japan,Australia,UK produces highest-rated movies on average (6.51,6.50,6.50 IMDb). But the gap between best and worst is 0.03 only.

In [31]:
# Top 10 Director by Average ROI

d_avg = df.groupby('Director').agg(
    Avg_ROI = ('ROI_pct','mean'),
    Count = ('MovieID','count')
).round(2).sort_values('Avg_ROI',ascending = False).head(10)
print('---Top 10 Director by Average ROI---')
print(d_avg)

---Top 10 Director by Average ROI
                   Avg_ROI  Count
Director                         
Kim Watkins         182.10   6657
Karen Smith         181.79   6689
Charles Lopez       181.39   6514
Robert Webster      181.24   6629
Raymond Morris      181.18   6635
Brandon Fuller      180.94   6750
Diana Davis         180.59   6699
Manuel Benton       180.51   6768
Roberto Underwood   180.36   6603
Suzanne Gonzalez    180.27   6591


**Top Director By ROI** — Kim Watkins, Karen Smith, Chales Lopez lead on avg ROI with 182.1%, 181.8%, 181.4% respectively - each with 6500+ movies.

In [10]:
# IMDb VS RT Correlation

corr_imdb_rt = df['IMDbRating'].corr(df['RottenTomatoesScore'])
print(f'IMDB Rating -> RT Correlation score : {corr_imdb_rt:.3f}')

IMDB Rating -> RT Correlation score : 0.829


**IMDb Vs RT Correlation** — Despite having different voting mechanisms (public votes vs. critic reviews), the two platforms share a strong correlation (0.829). If a movie is good, both tend to agree with it.

In [36]:
# Box Office Trend over Decade

B_D = df.groupby('Decade', observed=True)['Global_BoxOfficeUSD'].median().round(0)
print(f"---Median global box office by decade---")
print(B_D)

---Median global box office by decade---
Decade
1950s    7985272.0
1960s    8133277.0
1970s    8126271.0
1980s    8081277.0
1990s    8082126.0
2000s    8113213.0
2010s    8058511.0
2020s    8106695.0
Name: Global_BoxOfficeUSD, dtype: float64


**Box office over decades** — Median global box office is remarkably stable across all decades (~$8M)

#### Biggest Insight
* The single strongest business signal in this dataset is - **Opening Day Sales predict Total Box Ofiice with 0.941 correlation**. If a studio want to forecast a movie's lifetime revenue, they need to wait for one day. 